### PR TITLE
[BE] Test cases for login

### PR DESCRIPTION
**Pull request by MihaiDruta**
[BE] Test cases for login
https://trello.com/c/RsPpY9pu/107-be-test-cases-for-login

Added test cases for login

### CODE REVIEW

**Review by nspc97 — APPROVED** [review]

File: .env.test [hunk 1]
```diff
@@ -2,3 +2,9 @@
 KERNEL_CLASS='App\Kernel'
 APP_SECRET='s$cretf0rt3st'
 SYMFONY_DEPRECATIONS_HELPER=999999
+
+###> lexik/jwt-authentication-bundle ###
+JWT_SECRET_KEY=%kernel.project_dir%/config/jwt/private.pem
+JWT_PUBLIC_KEY=%kernel.project_dir%/config/jwt/public.pem
+JWT_PASSPHRASE=passphrase
+###< lexik/jwt-authentication-bundle ###
```
Comment (SpartakusMd) [inline .env.test]: Leave an empty line at the end.

File: src/DataFixtures/TypeFixtures.php [hunk 2]
```diff
@@ -12,7 +12,13 @@ class TypeFixtures extends Fixture
     public const TYPES = [
         'dt estimation',
```
Comment (SpartakusMd) [inline src/DataFixtures/TypeFixtures.php]: Make the words capitalized where needed. The shouldn't be all lowercase.

File: tests/LoginTest.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        return [
+            'header' => (array)$jwtHeader,
+            'payload' => (array)$jwtPayload,
+        ];
+    }
+
+    private function getUsernameRoles($token): array
+    {
+        $decodedToken = $this->tokenDecode($token);
```
Comment (SpartakusMd) [inline tests/LoginTest.php]: Use `Lexik\Bundle\JWTAuthenticationBundle\Encoder\JWTEncoderInterface::decode()`

**Review by SpartakusMd — CHANGES_REQUESTED** [review]
Travis is still failing.

File: src/DataFixtures/ActivityFixtures.php [hunk 4]
```diff
@@ -31,7 +31,7 @@ public function load(ObjectManager $manager): void
             $activity->setOwner($owner);
 
             /** @var ActivityType $type */
-            $type = $this->getReference('type_' . TypeFixtures::TYPES[rand(0, 2)]);
+            $type = $this->getReference('type_' . TypeFixtures::TYPES[rand(0, 8)]);
```
Comment (SpartakusMd) [inline src/DataFixtures/ActivityFixtures.php]: 8 will be undefined. The count starts from 0 not 1.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Pull request merged**